### PR TITLE
[LW-8108] disable tracking events

### PR DESCRIPTION
**Pull request by tommayeliog**
# Checklist

- [x] JIRA - [\<link>](https://input-output.atlassian.net/browse/LW-8108)
- [x] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

With this PR we are implementing a functionality to skip sending excluded events to PostHog.
.env variable "POSTHOG_EXCLUDE_EVENTS" holds comma separated event types.

## Testing
POSTHOG_EXCLUDE_EVENTS=
Set some events in POSTHOG_EXCLUDE_EVENTS and check if they are omitted from sending to PostHog.

Note: in POSTHOG_EXCLUDE_EVENTS you can define the whole event type or just the starting phrase.
e.g. "onboarding | new wallet" excludes both 'onboarding | new wallet | all done | go to my wallet | click' and 'onboarding | new wallet | analytics | agree | click'

## Screenshots
------

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2030/6185417646/index.html) for [3ca6a560](https://github.com/input-output-hk/lace/pull/478/commits/3ca6a560792ef5e17b2f2aeada5213e007ad3fa6)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 1      | 0       | 0     | 32    | ❌     |

### CODE REVIEW

File: apps/browser-extension-wallet/src/providers/AnalyticsProvider/analyticsTracker/AnalyticsTracker.test.ts [hunk 1]
```diff
@@ -124,6 +124,17 @@ describe('AnalyticsTracker', () => {
       expect(mockedPostHogClient.sendEvent).toHaveBeenCalledTimes(1);
       expect(userIdServiceMock.extendLifespan).toHaveBeenCalledTimes(1);
     });
+
+    it('should ommit sending excluded events', async () => {
```
Comment (marcin-mazurek) [inline apps/browser-extension-wallet/src/providers/AnalyticsProvider/analyticsTracker/AnalyticsTracker.test.ts:128]: How about if we set environment variable in the test itself or mock `POSTHOG_EXCLUDE_EVENTS` export from `apps/browser-extension-wallet/src/providers/AnalyticsProvider/postHog/config.ts` and have a few test cases to cover different scenarios (in particular edge cases such as empty value)?

**Review by marcin-mazurek — APPROVED** [review]

File: apps/browser-extension-wallet/src/providers/AnalyticsProvider/analyticsTracker/AnalyticsTracker.test.ts (hunk 1 above)
Comment (DominikGuzei) [reply apps/browser-extension-wallet/src/providers/AnalyticsProvider/analyticsTracker/AnalyticsTracker.test.ts:128]: Yes I like the idea of making the mocking/tests more versatile 👍

**Review by DominikGuzei — APPROVED** [review]
Great work 👍

**Review by marcin-mazurek — APPROVED** [review]

**Review by bslabiak — APPROVED** [review]

**Comment by tomislavhoracek** [comment]
@tommayeliog Could you please merge the latest `main` branch and fix conflicts? 🙏

**Review by bslabiak — APPROVED** [review]

**Review by danielmain — APPROVED** [review]

**Review by danielmain — APPROVED** [review]

**Comment by tomislavhoracek** [comment]
Thanks @danielmain  ❤️
